tests/integration/long/test_ssl.py

```diff
@@ -28,7 +28,7 @@
 
 if not hasattr(ssl, 'match_hostname'):
     try:
-        from backports.ssl_match_hostname import match_hostname
+        from ssl import match_hostname
```

Comment:
This seems like exactly the right idea.

I'll also note that ssl.match_hostname has been deprecated since Python 3.7 and is actually removed in 3.12 (more on that in [PYTHON-1331](https://datastax-oss.atlassian.net/browse/PYTHON-1331)) so these changes will likely be subsequently changed based on whatever happens with that ticket (which is next in my queue).  I very much like the other changes in this PR, though, so my inclination is to merge this now and just lay any fixes for PYTHON-1331 on top of this in their own PR.